Postgresql aggregate functions (#4695)
Codecademy/docs: content/postgresql/concepts/aggregate-functions/aggregate-functions.md

```

This query calculates the maximum amount spent on a single comodity in the `sales` table by multiplying the `Quantity` column with the `cost` column and then determines the maximum value.

## Min

```sql
SELECT MIN(Quantity * Cost) AS Min_Spent FROM sales;
```

The minimum amount spent on a single product in the `sales` table is calculated by this query, which multiplies the `Quantity` column with the `Cost` column and then finds the minimum value.

## Sum

```sql
SELECT SUM(Quantity * cost) AS Total_Spent FROM sales;
```

Calculating the total amount spent on all sales in the sales table involves multiplying the `Quantity` column with the `cost` column for each sale and then summing these values using this query, which labels the result as `Total_Spent`.

## Count

```sql
SELECT COUNT(*) FROM consumer;
```

This query returns the total count of records in the `sales` table.

## Avg

```sql
SELECT AVG(Quantity * Cost) AS Average_Spent FROM sales;
```

This query finds the average amount spent per sale in the sales table by calculating the product of the `Quantity` and `cost` columns for each sale and then averaging these values.

> ## GROUP BY
> 
> The `GROUP BY` clause in Postgresql is used to arrange identical data into groups. This clause is often combined with aggregate functions to perform operations on each group of data. Essentially, `GROUP BY` allows you to summarize data by grouping rows that have the same values in specified columns.

Reviewer: ```suggestion
### GROUP BY

The `GROUP BY` clause in PostgreSQL is used to arrange identical data into groups. This clause is often combined with aggregate functions to perform operations on each group of data. Essentially, `GROUP BY` summarizes data by grouping rows that have the same values in the specified columns.
```

Let's use the `sales` table to find out how much each customer spent in total.

```pseudo
SELECT Customer, SUM(Quantity * cost) AS Total_Spent
FROM sales
GROUP BY Customer;
```

> Breaking down the query:
> 
> - `SELECT Customer` specifies the column name to retrieve from the table.
> - `SUM(Quantity * cost) AS Total_Spent` multiplies the `Quantity` and `cost` columns for each sale to calculate the total amount spent by the customer, and names the result column `Total_Spent`.
> - `FROM sales` specifies the table from which the query retrieves the data.
> - `GROUP BY Customer` clause groups the results by the `Customer` column.

Reviewer: ```suggestion
The `GROUP BY` clause groups the set of results, including the total amount spent by each customer, by the `Customer` column.
```

> This is the out put of the above query:

Reviewer: ```suggestion
Here is the output of the above query:
```
